crates/bitwarden-crypto/src/service/mod.rs

```diff
@@ -0,0 +1,197 @@
+use std::sync::{Arc, RwLock};
+
+use crate::{AsymmetricKeyRef, Decryptable, Encryptable, KeyRef, SymmetricKeyRef, UsesKey};
+
+mod context;
+
+mod key_store;
+
+use context::ReadWriteGlobal;
+pub use context::{CryptoServiceContext, ReadOnlyGlobal};
+pub use key_store::create_key_store;
+use key_store::KeyStore;
+
+#[derive(Clone)]
+pub struct CryptoService<SymmKeyRef: SymmetricKeyRef, AsymmKeyRef: AsymmetricKeyRef> {
```

Comment:
**question**: Should this be called a service, or something more generic like key store? I think we can drop the references to service.

Reply:
We're already using `KeyStore` for the trait that only deals with the key storage, so we'd need to rename that or choose a different name. 

But yeah I'm happy to rename it if we have a better name.